src/qw/cli.py

```
@@ -1,30 +1,118 @@
+    logger.info(
+        ".qw directory is '{dir}'",
+        dir=qw_dir,
+    )
+    if os.path.isfile(qw_dir):
+        msg = ".qw file exists, which is blocking us from making a .qw directory. Please delete it!"
+        raise QwError(
+            msg,
+        )
+    elif not os.path.isdir(qw_dir):
+        os.mkdir(qw_dir)
+    elif not force:
+        msg = ".qw directory already exists! Use use existing configuration or use --force flag to reinitialize!"
+        raise QwError(
+            msg,
+        )
+    hur = qw.service.remote_address_to_host_user_repo(repo)
+    if hur is None:
+        msg = f"'{repo}' does not seem to be a valid repo url!"
+        raise QwError(
+            msg,
+        )
+    host = hur[0]
+    if service is None:
+        service = qw.service.hostname_to_service(host)
+    if service is None:
+        msg = f"'{host}' is not a service I know about!"
+        raise QwError(
+            msg,
+        )
+    logger.info(
+        "service, owner, repo: {service}, {owner}, {repo}",
+        service=str(service),
+        owner=hur[1],
+        repo=hur[2],
+    )
+    conf = {
+        "repo_url": repo,
+        "repo_name": hur[2],
+        "user_name": hur[1],
+        "service": str(service),
+    }
+    conf_file_name = os.path.join(qw_dir, "conf.json")
+    with open(conf_file_name, "w") as conf_file:
+        json.dump(conf, conf_file, indent=2)
 
 
 @app.command()
-def echo(
-        example: Annotated[str, typer.Argument(help="String to print to user")]
-) -> None:
-    logger.info("Echo: {example}", example=example)
+def check():
+    """Checks whether all the traceability information is present."""
+    conf = qw.service.get_configuration()
+    service = qw.service.get_service(conf)
+    print(conf)
+    print(service.get_issue(1).title())
 
 
 if __name__ == "__main__":
-    app()
+    try:
+        app()
+        sys.exit(0)
+    except QwError as e:
+        print(e)
+        sys.exit(2)
```

maybe we just don't catch the exceptions at all and let it bubble up to kill it with a non-zero exit status?

---

That leaves the user staring at a callstack, which I don't love. This is why I'm using a special exception type, so all the rest does dump the callstack.

---

yeah that's a fair reason.